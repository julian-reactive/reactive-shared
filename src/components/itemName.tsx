import React from 'react'

import Avatar from '@mui/material/Avatar'

export interface ItemNameProps {
  image?: string
  name: string
}
const ItemNameContainer: React.FC<ItemNameProps> = ({ image, name = '' }) => {
  return (
    <Avatar src={image} alt={name} variant='square'>
      {name.slice(0, 1).toUpperCase()}
    </Avatar>
  )
}

export const ItemName = React.memo(ItemNameContainer)
