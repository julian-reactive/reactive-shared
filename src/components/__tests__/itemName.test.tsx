import React from 'react'

import {render} from '@testing-library/react'
import { ItemName, ItemNameProps } from '../itemName'

const defaultProps: ItemNameProps = {
    name: 'test',
    image: 'test'   
}

const getRendered = (props = defaultProps) => render(<ItemName {...props} />)

describe('itemName', () => {
    it('should match snapshot', () => {
        const { container } = getRendered()
        expect(container.firstChild).toMatchSnapshot()
    })
})