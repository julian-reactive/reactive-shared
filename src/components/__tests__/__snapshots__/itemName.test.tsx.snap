// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemName should match snapshot 1`] = `
<div
  class="MuiAvatar-root MuiAvatar-square css-j9wkuc-MuiAvatar-root"
>
  <img
    alt="test"
    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
    src="test"
  />
</div>
`;
